app/screens/invite/invite.tsx

```diff
@@ -0,0 +1,419 @@
+
+        if (usersToAdd.length) {
+            const {members, error: addUsersToTeamError} = await addUsersToTeam(serverUrl, teamId, usersToAdd);
+
+            if (addUsersToTeamError) {
+                handleSendError();
+                return;
+            }
+
+            if (members) {
+                const membersWithError: Record<string, string> = {};
+                for (const {user_id, error} of members) {
+                    if (error) {
+                        membersWithError[user_id] = error.message;
+                    }
+                }
+
+                for (const userId of usersToAdd) {
+                    if (membersWithError[userId]) {
+                        notSent.push({userId, reason: membersWithError[userId]});
+                    } else {
+                        sent.push({userId, reason: formatMessage({id: 'invite.summary.member_invite', defaultMessage: 'Invited as a member of {teamDisplayName}'}, {teamDisplayName})});
+                    }
+                }
+            }
+        }
+
+        if (emails.length) {
+            const {members, error: sendEmailInvitesToTeamError} = await sendEmailInvitesToTeam(serverUrl, teamId, emails);
+
+            if (sendEmailInvitesToTeamError) {
+                handleSendError();
+                return;
+            }
+
+            if (members) {
+                const membersWithError: Record<string, string> = {};
+                for (const {email, error} of members) {
+                    if (error) {
+                        membersWithError[email] = isAdmin && error.server_error_id === ServerErrors.SEND_EMAIL_WITH_DEFAULTS_ERROR ? (
+                            formatMessage({id: 'invite.summary.smtp_failure', defaultMessage: 'SMTP is not configured in System Console'})
+                        ) : (
+                            error.message
+                        );
+                    }
+                }
+
+                for (const email of emails) {
+                    if (membersWithError[email]) {
+                        notSent.push({userId: email, reason: membersWithError[email]});
+                    } else {
+                        sent.push({userId: email, reason: formatMessage({id: 'invite.summary.email_invite', defaultMessage: 'An invitation email has been sent'})});
+                    }
+                }
+            }
+        }
+
+        setResult({sent, notSent});
+        setStage(Stage.RESULT);
+    };
+
+    useNavButtonPressed(CLOSE_BUTTON_ID, componentId, closeModal, [closeModal]);
+    useNavButtonPressed(SEND_BUTTON_ID, componentId, handleSend, [handleSend]);
+
+    useEffect(() => {
+        const buttons: NavButtons = {
+            leftButtons: [makeLeftButton(theme)],
+            rightButtons: stage === Stage.SELECTION ? [makeRightButton(theme, formatMessage, selectedCount > 0)] : [],
+        };
+
+        setButtons(componentId, buttons);
+    }, [theme, locale, componentId, selectedCount, stage, sendError]);
```

Comment:
To have this effect executed less times, you can change the dependency to `selectCount > 0` and stage === SELECTION

Reply:
Updated